### PR TITLE
feat(cli): new cli

### LINKED ISSUE #1186: Feedback for “Setup”
`YARN Berry not supported.` Which version numbers of yarn is not supported exactly?

### LINKED ISSUE #1214: package manager auto detection for typia
I'd like to add feature for cli tool to auto detect package manager.
I'll create for that

### PR DESCRIPTION
**Pull request by ryoppippi**
fixes: https://github.com/samchon/typia/issues/1214
fixes:  #1186

This PR is re-implementation of typia cli.

Looks beautiful so please check this out

To execute the new cli
```
pnpm i
pnpm build
pnpm cli 
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,8 +18,10 @@
     "test:bun": "bun run deploy/bun.ts",
     "test:template": "npm run --tag test --template",
     "-------------------------------------------------": "",
-    "build": "rimraf lib && tsc && rollup -c",
+    "build": "rimraf lib && tsc && tsc -p ./tsconfig.cli.json && rollup -c",
```
Comment (ryoppippi) [inline package.json:21]: in CJS environment, generating declarations fails for the cleye library, so I split the build command

**Review by samchon — COMMENTED** [review]
Any special reason of building ESM for CLI program?

Also, the different between previous CLI program and your PR is detecting which package manager being used. However, reviewing is too much hard for me because everything has been changed.  CLI program does not need to consider the tree shaking, but every namespaces are decomposed to individual exported functions. 

May I refactor your codes to the previous namespace structure?

**Comment by ryoppippi** [comment]
@samchon 
- the reason I use ESM for `bin/typia.mjs` is that esm can use top-level await. this is more readable.
- I'll refactor the code to use namespace import (instead of namespace). give me some times.
Thank you for reviewing as always

**Comment by samchon** [comment]
Oops, just hope to revive the namespace. Too much change for me.

**Comment by ryoppippi** [comment]
Oh, Okay, so I'll use namespace :)
However, this is a reminder that namespace is becoming less and less recommended, because namespace is a typescript-specific grammer .

**Comment by ryoppippi** [comment]
@samchon 
I refactored, but I'm sorry that I am not used to using namespace.
If my code is hard to read for you, please modify it.
And any questions are welcome

**Comment by samchon** [comment]
Will merge to a new branch, and publish with LLM function calling schema features as `6.10.0`.

it may be two weeks later.

**Review by samchon — COMMENTED** [review]
Will merge to a new branch, and publish with LLM function calling schema features as `6.10.0`.

it may be two weeks later.

**Comment by ryoppippi** [comment]
Oh you implemented for LLM? thank you

**Comment by samchon** [comment]
> Oh you implemented for LLM? thank you

Not yet but soon. It's the reason of 2 weeks later's v6.10 publlishing.

**Comment by ryoppippi** [comment]
@samchon 
If you want me to implement this llm modules(or transfer my `typiautil` implementation into `typia`), create an issue and assign me!
If you implement it, never mind

**Pull request merged**